docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md

```diff
@@ -150,8 +159,86 @@ The next step is to trigger the test by following command:
 ```
 $ saucectl run
 ```
+
 `saucectl` uploads `DemoApp.ipa` and `DemoAppUITests-Runner.ipa` to Sauce VM and executes XCUITest. You can review test results on Sauce UI.
 
+#### Sharding XCUITest
+
+`saucectl` supports running tests in parallel. To tests in parallel, configure `saucectl` as follows. See [yaml configuration](./xcuitest.md#shard) for more details.
+
+```
+suites:
+  - name: "sharded suites"
+    shard: concurrency
+    testListFile: test_classes.txt
+```
+
+The `test_classes.txt` can be generated by following these steps.
+
+1. Open the project in XCode and select test target
+2. Click the `Build Phases`
+3. In the `Build Phases`, click the `+`-sign
+4. Add the following script
+
+```
+# Set the path to the output file
+OUTPUT_FILE="${SRCROOT}/test_classes.txt"
+
+# Find all Swift files in the test target and exclude the ones you don't want
+TEST_FILES=$(find "${SRCROOT}/${TARGET_NAME}" -name "*.swift" -not -name "Info.plist")
+
+# Print the found test files
+echo "Found test files:"
+echo "$TEST_FILES"
+echo ""
+
+# Initialize an empty string to store class names
+CLASS_NAMES=""
+
+# Get the class names from the Swift files that contain test methods
+while IFS= read -r file; do
+    # Check if the file contains a test method
+    if grep -q "func test[A-Za-z0-9_]*()" "$file"; then
+        # Extract the class name and append it to the list
+        CLASS_NAME=$(grep -Eo "class [a-zA-Z_][a-zA-Z0-9_]*:*" "$file" | awk '{gsub(":", "", $2); print $2}')
+        CLASS_NAMES+="$CLASS_NAME"$'\n'
+    fi
+done <<< "$TEST_FILES"
+
+# Remove duplicate class names
+UNIQUE_CLASS_NAMES=$(echo "$CLASS_NAMES" | sort | uniq | grep -v '^$')
+
+# Write the unique class names to the output file
+echo "$UNIQUE_CLASS_NAMES" > "$OUTPUT_FILE"
+
+# Print the class names and the path to the output file
+echo "Test class names:"
+echo "$UNIQUE_CLASS_NAMES"
+echo ""
+echo "Test class list has been created at: ${OUTPUT_FILE}"
+```
+![add-build-script.png](/img/xcuitest/add-build-script.png)
+
+5. The script can be triggered with
+    * XCODE by:
+        * Going to `Product` > `Build For` > `Testing`
+            ![trigger-build-script.png](/img/xcuitest/trigger-build-script.png)
+        * Or triggering a Testing build by using `Command + Shift + U`
+    * Or through running this command
+```
+xcodebuild \
+  CODE_SIGN_IDENTITY="" \
+  CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO \
+  clean build-for-testing \
+  -workspace "{workspaceName}.xcworkspace" \
+  -scheme "{schemeName}" \
+  -sdk iphoneos
+```
+
+The generated `test_classes.txt` can be found here.
```

Comment:
Colon probably better, as I assume that you are referring to the picture after this?
```suggestion
The generated `test_classes.txt` can be found here:
```